src/subquery.js

```diff
@@ -0,0 +1,43 @@
+import _ from 'lodash/fp'
+import { getTypePropOrError } from './types'
+
+// A subquery (in contexture-client) is about taking the output of one search and makng it the input for another search.
+// This is an in memory, cross-database, "select in" join on sources that don't need to be relational.
+export default _.curry((types, from, fromPath, to, toPath) => {
+  let toNode = to.getNode(toPath)
+  let fromNode = from.getNode(fromPath)
+
+  // Dispatch on toNode when fromNode is markedForUpdate to trigger loading indicator immediately on toNode when fromNode updates
+  fromNode.onMarkForUpdate = () => {
+    // No need to await this, because we'll await the mutate action later
+    // We rely on the onMarkForUpdate on toNode to block this dispatch from completing until
+    // fromNode finishes, and since there is a debounce, it should not result in 2 searches.
+    // If the node fails validation after mutate, that would block this from running anyway because this shouldn't trigger an actual search.
+    to.dispatch({ type: 'all', path: toPath })
+  }
+
+  // Set validation dependency to block search, but uses onMarkForUpdate instead
+  //  so the toNode can be marked for update before fromNode resolves.
+  //  Validate blocks markedForUpdate but onMarkForUpdate does not.
+  toNode.onMarkForUpdate = () => fromNode.updatingPromise
+  // This version would not mark toNode for update until fromNode is done:
+  // toNode.validate = () => fromNode.updatingPromise.then(() => true)
+
+  // Gets results to use as input from the fromNode
+  let getSubqueryValues = getTypePropOrError(
+    types,
+    'getSubqueryValues',
+    fromNode
+  )
+  // Builds a mutate blob for the toNode based on the values from the previous method
+  let useSubqueryValues = getTypePropOrError(types, 'useSubqueryValues', toNode)
```

Comment:
Not sure if I like the `OrError` on this function. Does it throw? or does it just returns an error object?

Reply:
I think the opposite of this `OnError` naming convention is more widely used. The opposite being `maybe[name]` for the fault-tolerant function and `[name]` for the more strict function.

Reply:
Yes, it gets a type prop, but if it can't find it, it throws. Most of the time not throwing is the right behavior so we should probably optimize for the common use case.

Reply:
Yeah but that's not the normal expectation when writing code. Functions are supposed to throw when invalid values or something odd happens in the code. The ability to omit this error should be the exception rather than the norm.